tools/serverpod_cli/lib/src/logger/helpers/progress.dart

```diff
@@ -0,0 +1,156 @@
+import 'dart:async';
```

Comment:
If this code is taken from elsewhere, we may need to include their copyright.

Reply:
The foundation for this code was taken from [here](https://github.com/felangel/mason/blob/master/packages/mason_logger/lib/src/progress.dart) and is part of the [mason_logger package](https://pub.dev/packages/mason_logger).

The code is under the MIT license so I will include it. Details here: https://pub.dev/packages/mason_logger/license